src/Trie.mo

```diff
@@ -16,27 +16,90 @@
 /// The basic `Trie` operations consist of:
 /// - `put` - put a key-value into the trie, producing a new version.
 /// - `get` - get a key's value from the trie, or `null` if none.
+/// - `remove` - remove a key's value from the trie
 /// - `iter` - visit every key-value in the trie.
 ///
-/// The `put` and `get` operations work over `Key` records,
+/// The `put`, `get` and `remove` operations work over `Key` records,
 /// which group the hash of the key with its non-hash key value.
 ///
+/// Example:
 /// ```motoko
 /// import Trie "mo:base/Trie";
 /// import Text "mo:base/Text";
 ///
+/// // we do this to have shorter type names and thus
+/// // better readibility
 /// type Trie<K, V> = Trie.Trie<K, V>;
 /// type Key<K> = Trie.Key<K>;
 ///
-/// func key(t: Text) : Key<Text> { { key = t; hash = Text.hash t } };
+/// // we have to provide `put`, `get` and `remove` with 
+/// // a record of type Key<K> = { hash : Hash.Hash; key : K },
+/// // thus we define the following function that takes a value of type K 
+/// // (in this case `Text`) and returns a Key<K> record.
+/// func key(t: Text) : Key<Text> { { hash = Text.hash t; key = t } };
 ///
+/// // we start off by creating an empty Trie
```

Comment:
```suggestion
/// // we start off by creating an empty `Trie`
```

Reply:
for consistency this would need to be changed in the entire file then :D